### PR TITLE
Verify compute res - Add, Mul, Unconstrained

### PR DESCRIPTION
**Pull request by jkktom**
- Improved test cases for Add
- Negative Felt cases for Mul
- Unconstrained and Op1 cases
Please kindly discard former branch. I changed too much so just restarted.

### CODE REVIEW

File: pkg/vm/vm_test.go [hunk 1]
```diff
@@ -346,34 +348,168 @@ func TestInferOperandSub(t *testing.T) {
...
 
-	cellOp1 := &mem.Cell{
-		Accessed: true,
-		Value:    mem.MemoryValueFromInt(15),
-	}
+func TestComputeRes_Op1(t *testing.T) {
+	vm := defaultVirtualMachine()
+	instruction := Instruction{Res: Op1}
+
+	cellOp0 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromInt(10)}
```
Comment (rodrigo-pino) [inline pkg/vm/vm_test.go:367]: `cellOp0` is not used in this test, so no need to create it to test the function

File: pkg/vm/vm_test.go [hunk 2]
```diff
@@ -346,34 +348,168 @@ func TestInferOperandSub(t *testing.T) {
...
+}
 
-	expected := mem.MemoryValueFromMemoryAddress(
-		mem.NewMemoryAddress(2, 25),
-	)
+func TestComputeAddRes_AddrToFelt(t *testing.T) {
+	vm := defaultVirtualMachine()
+	instruction := Instruction{Res: AddOperands}
+
+	cellOp0 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 10))}
```
Comment (rodrigo-pino) [inline pkg/vm/vm_test.go:380]: You can use `MemoryValueFromSegmentAndOffset` method

File: pkg/vm/vm_test.go [hunk 3]
```diff
@@ -346,34 +348,168 @@ func TestInferOperandSub(t *testing.T) {
...
+	cellOp1 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromInt(15)}
+
+	res, err := vm.computeRes(&instruction, cellOp0, cellOp1)
+	require.NoError(t, err)
+	expected := mem.MemoryValueFromInt(25)
+	assert.Equal(t, expected, res)
+}
+
+// Felt should be Positive or Negative. Thus four test cases
+func TestComputeMulRes_PToPFelt(t *testing.T) {
```
Comment (rodrigo-pino) [inline pkg/vm/vm_test.go:427]:
```suggestion
func TestComputeMulResPosToPosFelt(t *testing.T) {
```

File: pkg/vm/vm_test.go [hunk 4]
```diff
@@ -346,34 +348,168 @@ func TestInferOperandSub(t *testing.T) {
...
+	_, err := vm.computeRes(&instruction, cellOp0, cellOp1)
+	require.Error(t, err) // Expecting an error since multiplying an address with a felt is not allowed
+}
+
+func TestComputeMulRes_FeltToAddr(t *testing.T) {
+	vm := defaultVirtualMachine()
+	instruction := Instruction{Res: MulOperands}
+
+	cellOp0 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromInt(15)}
+	cellOp1 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 10))}
```
Comment (rodrigo-pino) [inline pkg/vm/vm_test.go:497]: Use `MemoryFromSegmentAndOffset`

File: pkg/vm/vm_test.go [hunk 5]
```diff
@@ -346,34 +348,168 @@ func TestInferOperandSub(t *testing.T) {
 	assert.Equal(t, expectedOp0Cell, op0Cell)
 }
 
-func TestComputeAddRes(t *testing.T) {
+func TestComputeRes_Unconstrained(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (jkktom) [reply pkg/vm/vm_test.go:351]: Done.

File: pkg/vm/vm_test.go (hunk 3 above)
Comment (jkktom) [reply pkg/vm/vm_test.go:427]: Done. Applied to other function names too.

File: pkg/vm/vm_test.go [hunk 6]
```diff
@@ -346,34 +348,168 @@ func TestInferOperandSub(t *testing.T) {
...
 	vm := defaultVirtualMachine()
+	instruction := Instruction{Res: Unconstrained}
 
-	instruction := Instruction{
-		Res: AddOperands,
-	}
+	cellOp0 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromInt(10)}
+	cellOp1 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromInt(15)}
```
[the remark this answers is not in the document]
Comment (jkktom) [reply pkg/vm/vm_test.go:356]: done. Test goes with Nil as operands, and succeeds.

File: pkg/vm/vm_test.go (hunk 1 above)
Comment (jkktom) [reply pkg/vm/vm_test.go:367]: done. Cell Operand 0 set to nil, the test succeeds.

**Pull request closed without merging**